src/latest/dbt_yml_files.py

```diff
@@ -0,0 +1,1009 @@
+        ...,
+        description="The name of the model whose behaviour you are testing. Does not need to be wrapped in a ref.",
+        examples=["my_model"],
+    )
+    overrides: Optional[Overrides] = ""
+
+
+class BooleanOrJinjaString(RootModel[Union[JinjaString, bool]]):
+    root: Union[JinjaString, bool]
+
+
+class Constraint(BaseModel):
+    name: Optional[str] = ""
+    type: str
+    columns: Optional[StringOrArrayOfStrings] = ""
+    expression: Optional[str] = ""
+    warn_unenforced: Optional[BooleanOrJinjaString] = ""
+    warn_unsupported: Optional[BooleanOrJinjaString] = ""
+
+
+class Constraints(RootModel[List[Constraint]]):
+    root: List[Constraint]
+
+
+class ConversionTypeParams(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    base_measure: MetricInputMeasure
+    calculation: Optional[Calculation] = Calculation.conversion_rate
+    constant_properties: Optional[List[ConstantProperty]] = ""
+    conversion_measure: MetricInputMeasure
+    entity: str = Field(..., description="The entity to calculate over")
+    window: Optional[str] = ""
+
+
+class ConversionMetricTypeParams(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    conversion_type_params: Optional[ConversionTypeParams] = ""
+
+
+class CumulativeMetricTypeParams(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    grain_to_date: Optional[TimeGranularity] = TimeGranularity.month
+    measure: Optional[MetricInputMeasure] = ""
+    window: Optional[str] = ""
+
+
+class DerivedMetricTypeParams(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    expr: Optional[str] = ""
+    metrics: Optional[List[MetricInputSchema]] = ""
+
+
+class DimensionTypeParams(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    time_granularity: TimeGranularity
+    validity_params: Optional[ValidityParams] = ""
+
+
+class FreshnessRules(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    count: NumberOrJinjaString
+    period: FreshnessPeriod
+
+
+class Grants(RootModel[Dict[str, StringOrArrayOfStrings]]):
+    root: Dict[str, StringOrArrayOfStrings] = Field(
+        ...,
+        description="grant config. each key is a database permission and the value is the grantee of that permission",
+    )
+
+
+class IncludeExclude(BaseModel):
+    exclude: Optional[StringOrArrayOfStrings] = ""
+    include: Optional[StringOrArrayOfStrings] = ""
+
+
+class Measure(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+        regex_engine="python-re",
+    )
+    name: str = Field(..., pattern="(?!.*__).*^[a-z][a-z0-9_]*[a-z0-9]$")
+    description: Optional[str] = ""
+    agg: Agg
+    agg_params: Optional[AggregationTypeParams] = ""
+    agg_time_dimension: Optional[str] = Field(
+        None, pattern="(?!.*__).*^[a-z][a-z0-9_]*[a-z0-9]$"
+    )
+    create_metric: Optional[bool] = True
```

Comment:
I think the default should be None? https://github.com/dbt-labs/ai-codegen-api/blob/main/src/codegen/llm_output/semantic_model.py#L156